libvmaf/src/feature/cambi.c

```diff
@@ -719,9 +734,33 @@ static FORCE_INLINE inline double weight_scores_per_scale(double *scores_per_sca
     return score / normalization;
 }
 
-static int cambi_score(VmafPicture *pics, uint32_t *mask_dp, uint16_t window_size, double topk,
-                       const uint16_t *tvi_for_diff, float *c_values, uint16_t *c_values_histograms, 
-                       uint8_t *filter_mode_histogram, uint16_t *filter_mode_buffer, double *score) {
+void write_uint16(FILE *file, uint16_t v) {
+    fwrite((void*)(&v), sizeof(v), 1, file);
+}
+
+static void dump_c_values(float *c_values, int width, int height, int scale, int window_size, char *dir_path) {
+    int max_diff_weight = g_diffs_weights[0];
+    for (int i = 0; i < NUM_DIFFS; i++) {
+        if (g_diffs_weights[i] > max_diff_weight) {
+            max_diff_weight = g_diffs_weights[i];
+        }
+    }
+    char path[1024];
+    snprintf(path, sizeof(path), "%s/cambi_heatmap_scale_%d.yuv", dir_path, scale);
+    int max_c_value = max_diff_weight * window_size * window_size / 4;
+    int scaling_value = (1 << 16) - 1;
+    FILE *file = fopen(path, "a");
+    for (int i = 0; i < height; i++) {
+        for (int j = 0; j < width; j++) {
+            uint16_t val = scaling_value * c_values[i * width + j] / max_c_value;
```

Comment:
Might be faster to compute `max_c_value_inv = 1/max_c_value` outside the nested loops and use `* max_c_value_inv` inside the nested loops

Reply:
True, we can even do `scaling_value = ((1 << 16) - 1)/max_c_value` since we're multiplying by that anyway.